src/bin.rs

```diff
@@ -8,35 +9,91 @@ use crate::item::Item;
     pub fn new(dims: [Dimension; 3]) -> Self {
         Self {
-            block: Block::new(dims[0], dims[1], dims[2]),
+            blocks: vec![Block::new(dims[0], dims[1], dims[2])],
+            items: vec![],
         }
     }
 
-    /// Returns whether or not the Bin's dimensions can emcompass or match the Bin.
-    pub fn does_item_fit(&self, item: &Item<'_>) -> bool {
-        self.block.does_it_fit(&item.block)
-    }
+    /**
+    Returns whether or not the Bin's dimensions can emcompass or match the item.
 
-    /// Returns the remaining bins after the item has been added to the current bin.
-    /// Returns None if the item is too big to fit into the bin.
-    pub fn best_fit(self, item: &Item<'_>) -> Option<Vec<Bin>> {
-        self.block
-            .best_fit(&item.block)
-            .map(|blocks| blocks.into_iter().map(|block| Bin::from(block)).collect())
+    An item that fits into a bin:
+    ```rust
+        use bin_packer_3d::bin::Bin;
+        use bin_packer_3d::item::Item;
+        let item = Item::new("item1", [1.0, 2.0, 3.0]);
+        let bin = Bin::new([1.0, 2.0, 3.0]);
+        assert!(bin.fits(&item));
+    ```
+    An item that does not fit into a bin:
+    ```rust
+        use bin_packer_3d::bin::Bin;
+        use bin_packer_3d::item::Item;
+        let item = Item::new("item2", [4.0, 12.0, 14.0]);
+        let bin = Bin::new([3.0, 12.0, 14.0]);
+        assert!(!bin.fits(&item));
+    ```
+    **/
+    pub fn fits(&self, item: &Item<'_>) -> bool {
+        self.blocks
+            .iter()
+            .any(|block| block.does_it_fit(&item.block))
     }
-}
 
-impl From<Block> for Bin {
-    fn from(block: Block) -> Self {
-        Bin::new([block.dims[0], block.dims[1], block.dims[2]])
+    /**
+     Add the item to the bin
+     Returns None if the item cannot be packed into the bin.
+
+    ```rust
+        use bin_packer_3d::bin::Bin;
+        use bin_packer_3d::item::{Item, ItemId};
+
+        let item_1 = Item::new("item1", [24.0, 10.0, 2.0]);
+        let item_2 = Item::new("item2", [24.0, 10.0, 2.0]);
+        let item_3 = Item::new("item3", [24.0, 10.0, 2.0]);
+        let mut bin = Bin::new([24.0, 10.0, 4.0]);
+        assert!(bin.try_packing(item_1).is_some());
+        assert!(bin.try_packing(item_2).is_some());
+        assert_eq!(bin.try_packing(item_3), None);
+        assert_eq!(
+            bin.items
+                .into_iter()
+                .map(|item| item.id)
+                .collect::<Vec<&ItemId>>(),
+            vec!["item1", "item2"]
+        );
+    ```
+    **/
+    pub fn try_packing(&mut self, item: Item<'a>) -> Option<()> {
+        let block_to_pack_index =
+            self.blocks
+                .iter()
+                .enumerate()
+                .find_map(|(block_index, block)| {
+                    if block.does_it_fit(&item.block) {
+                        Some(block_index)
+                    } else {
+                        None
+                    }
```

Comment:
Seems like this would be a good use case for [bool_ext](https://crates.io/crates/bool_ext), right?

Reply:
Yup! :)  Exactly the use case `bool_ext` is designed for.